### PR TITLE
[BUG FIX] - Making JobInnerError object recursively defined

### PR DESCRIPTION
**Pull request by ro-joowan**
* Based on this PR: Azure/azure-rest-api-specs#1627

* We are essentially adding one more property to the JobInnerError object (recursively) in order to capture the additional error information that's returned by the service

* This added property was tested successfully by a team member

* Two file changes are in a previous PR (https://github.com/Azure/azure-sdk-for-net/pull/3634): SubmitGetListCancelTest.json and DataLakeStoreClientTests.cs.  I just needed to push this PR as soon as possible.

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Customizations/DataLakeAnalyticsCustomizationHelper.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ internal static class DataLakeAnalyticsCustomizationHelper
         /// This constant is used as the default package version to place in the user agent.
         /// It should mirror the package version in the project.json file.
         /// </summary>
-        internal const string PackageVersion = "3.1.1-preview";
+        internal const string PackageVersion = "3.1.2-preview";
```
Comment (shahabhijeet) [inline src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Customizations/DataLakeAnalyticsCustomizationHelper.cs]: @ro-joowan can you explain what is this package number and what project.json file are you referring to.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Customizations/DataLakeAnalyticsCustomizationHelper.cs (hunk 1 above)
Comment (ro-joowan) [reply src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Customizations/DataLakeAnalyticsCustomizationHelper.cs]: @shahabhijeet Couldn't find a project.json file in the src.  Ben might be referring to the csproj file (https://github.com/ro-joowan/azure-sdk-for-net/blob/341b0bbea8d9110f2047f9d28150a4cc95ef34fc/src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics.csproj), as this is one of the three places I've been told to update the version number -- the other place being in AssemblyInfo.cs

As for its purpose, I found this (https://github.com/ro-joowan/azure-sdk-for-net/blob/341b0bbea8d9110f2047f9d28150a4cc95ef34fc/src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Customizations/DataLakeAnalyticsAccountManagementClient.Customizations.cs):

/// <param name='userAgentAssemblyVersion'>
/// Optional. The version string that should be sent in the user-agent header for all requests. The default is the current version of the SDK.
/// </param>

It looks like this custom method is a way for the user to set the SDK version.

Comment (ro-joowan) [reply src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Customizations/DataLakeAnalyticsCustomizationHelper.cs]: @shahabhijeet And so this PackageVersion is used as the default SDK version in the event that the user does not provide a custom version (https://github.com/ro-joowan/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Customizations/DataLakeAnalyticsCustomizationHelper.cs#L29)

**Comment by ro-joowan** [comment]
@shahabhijeet Friendly ping on this :)

File: src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Customizations/DataLakeAnalyticsCustomizationHelper.cs (hunk 1 above)
Comment (shahabhijeet) [reply src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Customizations/DataLakeAnalyticsCustomizationHelper.cs]: Let's revisit this in the next release.
Will allow now.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**